src/server/frontend_wayland/text_input_v1.cpp

```diff
@@ -427,16 +430,13 @@ void TextInputV1::commit_state(uint32_t client_serial)
     if (pending_state && current_surface)
     {
         auto const hub_serial = ctx->text_input_hub->set_handler_state(handler, on_new_input_field, *pending_state);
+        pending_state_used = true;
         state_serials.push_back({client_serial, hub_serial});
         while (state_serials.size() > max_remembered_serials)
         {
             state_serials.pop_front();
         }
     }
-    else
-    {
-        ctx->text_input_hub->deactivate_handler(handler);
-    }
```

Comment:
This is the other substantive change: committing the state can no longer call `deactivate_handler()`